internal/server/errors.go

```diff
@@ -47,16 +48,13 @@ func (a *API) sendAPIError(c *gin.Context, err error) {
 
 	case errors.Is(err, internal.ErrBadRequest):
 		resp.Code = http.StatusBadRequest
-		resp.Message = err.Error()
+		resp.Message = trimErrorMsg(internal.ErrBadRequest, err)
 	case errors.Is(err, internal.ErrNotImplemented):
 		resp.Code = http.StatusNotImplemented
-		resp.Message = internal.ErrNotImplemented.Error()
+		resp.Message = trimErrorMsg(internal.ErrNotImplemented, err)
 	case errors.Is(err, internal.ErrBadGateway):
 		resp.Code = http.StatusBadGateway
-		resp.Message = err.Error()
-	case errors.Is(err, (*validator.InvalidValidationError)(nil)):
-		resp.Code = http.StatusBadRequest
-		resp.Message = err.Error()
```

Comment:
I tested this to confirm, but this is an incorrect use of `errors.Is`. I didn't realize this either until I was pairing with Elizabeth last week.

`errors.Is` only works with error variables. It's like the `==` operation, and `err` here isn't going to be `nil`.

This would need to use `errors.As` to do a type check. But I think even that isn't necessarily correct. `InvalidValidationError` is returned when an invalid argument is passed to `Validate`. If we pass  an invalid argument that's a programming error, and I think should probably be a 500 status code. So by removing this case we should end up with a 500 status code.

Reply:
the goal of the code here was to return StatusBadRequest when the request failed validation. We should try to accomplish that.

Reply:
That is already handled in the case above `case errors.As(err, validationErrors)`. Validation errors return `validator.ValidationErrors`.

`validator.InvalidValidationError` is documented as :

> // InvalidValidationError describes an invalid argument passed to
> // `Struct`, `StructExcept`, StructPartial` or `Field`

That sounds like a programming error , and should be a 500. So I think removing this is correct.